fedimint-core/src/util/mod.rs

```diff
@@ -159,6 +159,22 @@ impl SafeUrl {
     pub fn join(&self, input: &str) -> Result<SafeUrl, ParseError> {
         self.0.join(input).map(SafeUrl)
     }
+
+    pub fn is_onion_address(&self) -> bool {
+        let onion_suffix = ".onion";
+        let host = self
+            .host_str()
+            .expect("It should've asserted for `host` on construction");
+        let suffix_start = host
+            .len()
+            .checked_sub(onion_suffix.len())
+            .expect("It should not fail!");
```

Comment:
Why shouldn't this fail? E.g. `a.io` is a valid domain shorter than `.onion`. Could you just use [`str::ends_with`](https://doc.rust-lang.org/std/primitive.str.html#method.ends_with)?

Reply:
I was trying to follow what arti's team did, but ends_with will make it simpler for sure.

Reply:
I've fixed it to use .ends_with() instead.